### DISCUSSION TITLE
Can't compile dioxus due to wasm-bindgen-cli version

### DISCUSSION THREAD

**Discussion by MrTitanHearted** [wasm-bindgen/wasm-bindgen#3515] [Q&A]
### Dioxus Web app is not compiling

This is what compiler is giving me:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: 

it looks like the Rust project used to create this wasm file was linked against
version of wasm-bindgen that uses a different bindgen format than this binary:

  rust wasm file schema version: 0.2.87
     this binary schema version: 0.2.86

Currently the bindgen format is unstable enough that these two schema versions
must exactly match. You can accomplish this by either updating this binary or
the wasm-bindgen dependency in the Rust project.

You should be able to update the wasm-bindgen dependency with:

    cargo update -p wasm-bindgen --precise 0.2.86

don't forget to recompile your wasm file! Alternatively, you can update the
binary with:

    cargo install -f wasm-bindgen-cli --version 0.2.87

if this warning fails to go away though and you're not sure what to do feel free
to open an issue at https://github.com/rustwasm/wasm-bindgen/issues!
', C:\Users\abduq\.cargo\registry\src\github.com-1ecc6299db9ec823\dioxus-cli-0.3.2\src\builder.rs:133:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: 🚫 Serving project failed: Build Failed: Bindgen build failed!
This is probably due to the Bindgen version, dioxus-cli using `0.2.81` Bindgen crate
```

I tried to use wasm-bindgen-cli all v0.2.81, v0.2.86 and v0.2.87. It did not work in all cases

**Comment by daxpedda** [comment]
Could you share the error message when you try to run `wasm-bindgen-cli` v0.2.87?
According to the error message, that should make it work.

**Reply by MrTitanHearted** [reply]
I tried, it didn't work
what made it work was including:
```
wasm-bindgen = "=0.2.86"
```
however, when I imported other crates such as web-sys, it again started to show this error of mismatching versions, but this time with web-sys that uses new version!

**Reply by daxpedda** [reply]
This is expected, if you use `cargo update -p wasm-bindgen --precise 0.2.86`, like the error message tells you to, it would be the same thing.

Generally speaking the error message tells you everything that you need to know as long as you understand how `Cargo.lock` works and how dependency versions are selected.

The information you are sharing is quite little, there's not much to go on to help you. It would be very helpful if you share a larger example of your problem and at a minimum share the error message.
